srtcore/channel.cpp

```diff
@@ -581,15 +653,37 @@ EReadStatus CChannel::recvfrom(sockaddr* addr, CPacket& packet) const
     }
 
 #ifndef _WIN32
+    msghdr mh; // will not be used on failure
+
     if (select_ret > 0)
     {
-        msghdr mh;
         mh.msg_name = addr;
         mh.msg_namelen = m_iSockAddrSize;
         mh.msg_iov = packet.m_PacketVector;
         mh.msg_iovlen = 2;
+
+#ifdef SRT_ENABLE_PKTINFO
+        if (!m_bBindMasked)
+        {
+            // We don't need ancillary data - the source address
+            // will always be the bound address.
+            mh.msg_control = NULL;
+            mh.msg_controllen = 0;
+        }
+        else
+        {
+            // Extract the destination IP address from the ancillary
+            // data. This might be interesting for the connection to
+            // know to which address the packet should be sent back during
+            // the handshake and then addressed when sending during connection.
+            mh.msg_control = m_acCmsgRecvBuffer;
+            mh.msg_controllen = sizeof m_acCmsgRecvBuffer;
+        }
+#else
         mh.msg_control = NULL;
         mh.msg_controllen = 0;
+#endif
```

Comment:
Not quite similar, but minimizes the code
```
mh.msg_control = NULL;
mh.msg_controllen = 0;

#ifdef SRT_ENABLE_PKTINFO
if (m_bBindMasked)
{
    // Extract the destination IP address from the ancillary
    // data. This might be interesting for the connection to
    // know to which address the packet should be sent back during
    // the handshake and then addressed when sending during connection.
    mh.msg_control = m_acCmsgRecvBuffer;
    mh.msg_controllen = sizeof m_acCmsgRecvBuffer;
}
#endif
```

Reply:
Maybe it simplifies, but this one does two assignment, and can only hope that the compiler is intelligent enough to eliminate the first one as ineffective. It wouldn't be so significant, unless the fact that it likes on a critical path.